backend/comballoc.ml

```diff
@@ -28,7 +28,32 @@ type allocation_state =
     No_alloc
   | Pending_alloc of pending_alloc
 
-let rec combine i allocstate =
+let dump_formatter : Format.formatter option Lazy.t = lazy (
+    match !Flambda_backend_flags.dump_comballoc_path with
+    | "" -> None
+    | dir ->
+      let file = Sys.argv |> Array.to_list |> String.concat " " |> Digest.string |> Digest.to_hex in
+      let path = Filename.concat dir file in
+      let chan = open_out path in
+      let ppf = Format.formatter_of_out_channel chan in
+      at_exit (fun () -> close_out_noerr chan);
+      Some ppf)
+
```

Comment:
We can use the existing `ppf_dump` in `Asmgen` and just pass it to `Comballoc`.  I don't think we need the new flag `-dcomballoc-path`. The printing can be enabled using the existing `-dcombine` flag, it will just print a few extra lines. I don't think anything relies on the format of the output of this command.  To print to a file, we already have `-dump-into-file` flag that would print to `foo.cmx.dump` for example when the compilation command had `-o foo.cmx`